### PR TITLE
fix(createComponent): change to TS and storybook new version

### PR DESCRIPTION
**Pull request by GaroGabrielyan**

### CODE REVIEW

File: scripts/createComponent.js [hunk 1]
```diff
@@ -351,68 +329,92 @@ const createStoryFiles = async (storyData, { name, level, ...restCmpData }) => {
...
+                const beforeSeparator = match.substring(0, lastIndex + 1);
+                const afterSeparator = match.substring(lastIndex + 1);
+                const newMatch = beforeSeparator + exportStatement + afterSeparator;
+                const newData = data.replace(match, newMatch);
+                const formattedData = prettier.format(newData, { ...prettierConfig, parser: 'typescript' });
+                await fs.writeFile(indexTsPath, formattedData);
+            }
+        }
     } catch (error) {
         console.log(error);
```
Comment (NarekArshakyan) [inline scripts/createComponent.js:419]: we  need this  console.log ?

File: scripts/createComponent.js [hunk 2]
```diff
@@ -7,6 +7,11 @@ import process from 'process';
 import ora from 'ora';
 import prettier from 'prettier';
 
+const firstLetterChange = (str, toUpperCase = true) => {
+    const firstChar = str.charAt(0);
```
Comment (NarekArshakyan) [inline scripts/createComponent.js:11]: `const firstChar = str[0]`

File: scripts/createComponent.js [hunk 3]
```diff
@@ -34,72 +39,93 @@ const messages = {
...
+const pathToLib = ['..', 'src', 'lib'];
+
+const generateCmpTemplate = ({ name, description, props, isWithForwardRef }) => {
+    const InterfaceName = `I${name}Props`;
+
+    return `
+    import React${isWithForwardRef ? ', { forwardRef }' : ', { FC }'} from 'react';
 
+    // Styles
     import 'src/assets/styles/globalStyling.scss';
```
Comment (NarekArshakyan) [inline scripts/createComponent.js:51]: ?

File: scripts/createComponent.js [hunk 4]
```diff
@@ -182,7 +208,7 @@ const askQuestions = () => {
             name: 'files',
             type: 'input',
             message:
-                "If you need extra .js files in the component folder type files names separated by ',' else leave the input empty: ",
+                "If you need extra .ts files in the component folder type files names separated by ',' else leave the input empty: ",
             prefix: '[?]',
             filter: (value) => (value ? value.split(',').map((prop) => prop.trim()) : []),
             validate: async (files) => {
```
Comment (NarekArshakyan) [inline scripts/createComponent.js:214]: what is async doing here?

File: scripts/createComponent.js [hunk 5]
```diff
@@ -7,6 +7,11 @@ import process from 'process';
 import ora from 'ora';
 import prettier from 'prettier';
 
+const firstLetterChange = (str, toUpperCase = true) => {
+    const firstChar = str.charAt(0);
+    return (toUpperCase ? firstChar.toUpperCase() : firstChar.toLowerCase()) + str.slice(1);
```
Comment (NarekArshakyan) [inline scripts/createComponent.js:12]: ` const getFirstChar=toUpperCase ? firstChar.toUpperCase() : firstChar.toLowerCase()   
 str.replace(str[0],getFirstChar)
`

**Review by hamikhambardzumyan — APPROVED** [review]
Looks good to me

**Pull request merged**